sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

```diff
@@ -0,0 +1,670 @@
+#### Table1 - Valid values of `N` if cache-hints are used or `pred` parameter is passed:
+| sizeof(`T`) | Valid values of `N` | Special case - PVC only - the maximal `N`: requires bigger alignment: 8 or more |
+|---|--------------------------------|-----|
+| 1 | 4, 8, 12, 16, 32, 64, 128, 256 | 512 |
+| 2 | 2, 4, 6, 8, 16, 32, 64, 128 | 256 |
+| 4 | 1, 2, 3, 4, 8, 16, 32, 64 | 128 |
+| 8 | 1, 2, 3, 4, 8, 16, 32 | 64 |
+
+## block store(...) - fast store to a contiguous memory block
+```C++
+namespace sycl::ext::intel::esimd {
+// block store to USM memory.
+template <typename T, int N, typename PropertyListT = empty_properties_t>
+/*usm-bs-1*/ void block_store(T* ptr, simd<T, N> vals, PropertyListT props={});
+/*usm-bs-2*/ void block_store(T* ptr, size_t byte_offset, simd<T, N> vals, PropertyListT props={});
+/*usm-bs-3*/ void block_store(T* ptr, simd<T, N> vals, simd_mask<1> pred, PropertyListT props={});
+/*usm-bs-4*/ void block_store(T* ptr, size_t byte_offset, simd<T, N> vals, simd_mask<1> pred, PropertyListT props={});
+
+// block store to device accessor. OffsetT is uint64_t in `stateless` mode(default), and uint32_t in `stateful`
+template <typename T, int N, typename AccessorT, typename PropertyListT = empty_properties_t>
+/*acc-bs-1*/ void block_store(AccessorT acc, OffsetT byte_offset, simd<T, N> vals, PropertyListT props = {});
+/*acc-bs-2*/ void block_store(AccessorT acc, simd<T, N> vals, PropertyListT props = {});
+/*acc-bs-3*/ void block_store(AccessorT acc, OffsetT byte_offset, simd<T, N> vals, simd_mask<1> pred, PropertyListT props = {});
+/*acc-bs-4*/ void block_store(AccessorT acc, simd<T, N> vals, simd_mask<1> pred, PropertyListT props = {});
+
+// block store to local accessor (SLM).
+template <typename T, int N, typename AccessorT, typename PropertyListT = empty_properties_t>
+/*lacc-bs-1*/ void block_store(AccessorT lacc, uint32_t byte_offset, simd<T, N> vals, PropertyListT props={});
+/*lacc-bs-2*/ void block_store(AccessorT lacc, simd<T, N> vals, PropertyListT props={});
+/*lacc-bs-3*/ void block_store(AccessorT lacc, uint32_t byte_offset, simd<T, N> vals, simd_mask<1> pred, PropertyListT props={});
+/*lacc-bs-4*/ void block_store(AccessorT lacc, simd<T, N> vals, PropertyListT props={});
+void block_store(AccessorT lacc, simd<T, N> vals, simd_mask<1> pred, PropertyListT props={});
+
+// block store to SLM (Shared Local Memory).
+template <typename T, int N, typename PropertyListT = empty_properties_t>
+/*slm-bs-1*/ void slm_block_store(uint32_t byte_offset, simd<T, N> vals, simd_mask<1> pred, PropertyListT props={});
+/*slm-bs-2*/ void slm_block_store(uint32_t byte_offset, simd<T, N> vals, PropertyListT props={});
+}
+```
+### Description
+`(usm-bs-*)`: Stores `vals` to a contiguous global memory block referenced by the USM pointer `ptr` optionally adjusted by `byte_offset`.  
+`(acc-bs-*)`, `(lacc-bs-*)`: Stores `vals` to a contiguous global memory block referenced by the accessor optionally adjusted by `byte_offset`.  
```

Comment:
for the lacc-bs case, is it accurate to say it is global memory? should we reword it?